src/tox/interpreters/unix.py

```diff
@@ -8,12 +10,15 @@
 
 @tox.hookimpl
 def tox_get_python_executable(envconfig):
+    """
+    Return path to specified interpreter.
+    If not available, exit(1).
+    """
     spec, path = base_discover(envconfig)
-    if path is not None:
+
+    if path is None:
+        print('envname {} not found'.format(envconfig.envname))
+        sys.exit(1)
```

Comment:
You're not supposed to use sys.exit here, instead raise an exception.

Reply:
Can you say which exception could be raised to give a concise error message back to the user?

I have tried raising tox.exception.UnsupportedInterpreter but that presented a lot of error text to the user.

Reply:
Would need to investigate, but that does seem the right exception type. Also, I don't think this should be handled here, but higher up in the call chain.